model/generate_test.go

```diff
@@ -1664,17 +1664,30 @@ func TestSimulateNewDependencyGraph(t *testing.T) {
 	require.NoError(t, generatorTask.Insert())
 
 	t.Run("CreatesCycle", func(t *testing.T) {
+		// Generator generates: [ A (depends on B) , B (depends on) C), C (depends on A) ]
+		// the graph:
+		//   +-----+            +-----+
+		//   |  A  |            |  C  |
+		//   +-----+ <--------- +-----+
+		//            \         ^
+		//             v       /
+		//              +-----+
+		//              |  B  |
+		//              +-----+
+
 		project := &Project{
 			BuildVariants: []BuildVariant{
 				{Name: "bv0", Tasks: []BuildVariantTaskUnit{
-					{Name: "generated", Variant: "bv0", DependsOn: []TaskUnitDependency{{Name: "dependedOn", Variant: "bv0"}}},
-					{Name: "dependedOn", Variant: "bv0", DependsOn: []TaskUnitDependency{{Name: "generator", Variant: "bv0"}}},
```

Comment:
Note: This setup is recreated as a non error case and explained more in the "Dependent on Generator" test case.